src/lib/core/datetime/native-date-adapter.spec.ts

```diff
@@ -0,0 +1,219 @@
+      '25日', '26日', '27日', '28日', '29日', '30日', '31日'
+    ]);
+  });
+
+  it('should get long day names', () => {
+    expect(adapter.getDayNames('long')).toEqual([
+      'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'
+    ]);
+  });
+
+  it('should get short day names', () => {
+    expect(adapter.getDayNames('short')).toEqual(['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat']);
+  });
+
+  it('should get narrow day names', () => {
+    expect(adapter.getDayNames('narrow')).toEqual(['S', 'M', 'T', 'W', 'T', 'F', 'S']);
+  });
+
+  it('should get day names in a different locale', () => {
+    expect(adapter.getDayNames('long', 'ja-JP')).toEqual([
+        '日曜日', '月曜日', '火曜日', '水曜日', '木曜日', '金曜日', '土曜日'
+    ]);
+  });
+
+  it('should get year name', () => {
+    expect(adapter.getYearName(new Date(2017, 0, 1))).toBe('2017');
+  });
+
+  it('should get year name in a different locale', () => {
+    expect(adapter.getYearName(new Date(2017, 0, 1), 'ja-JP')).toBe('2017年');
+  });
+
+  it('should get long month and year name', () => {
+    expect(adapter.getMonthYearName(new Date(2017, 0, 1), 'long')).toBe('January 2017');
+  });
+
+  it('should get short month and year name', () => {
+    expect(adapter.getMonthYearName(new Date(2017, 0, 1), 'short')).toBe('Jan 2017');
+  });
+
+  it('should get narrow month and year name', () => {
+    expect(adapter.getMonthYearName(new Date(2017, 0, 1), 'narrow')).toBe('J 2017');
+  });
+
+  it('should get month and year name in a different locale', () => {
+    expect(adapter.getMonthYearName(new Date(2017, 0, 1), 'long', 'ja-JP')).toBe('2017年1月');
+  });
+
+  it('should get first day of week', () => {
+    expect(adapter.getFirstDayOfWeek()).toBe(0);
+  });
+
+  it('should create Date', () => {
+    expect(adapter.create(2017, 0, 1)).toEqual(new Date(2017, 0, 1));
+  });
+
+  it('should create Date with month and date overflow', () => {
+    expect(adapter.create(2017, 12, 32)).toEqual(new Date(2018, 1, 1));
+  });
+
+  it('should create Date with month date underflow', () => {
+    expect(adapter.create(2017, -1, 0)).toEqual(new Date(2016, 10, 30));
+  });
+  
+  it('should create Date with low year number', () => {
+    expect(adapter.create(-1, 0, 1).getFullYear()).toBe(-1);
+    expect(adapter.create(0, 0, 1).getFullYear()).toBe(0);
+    expect(adapter.create(50, 0, 1).getFullYear()).toBe(50);
+    expect(adapter.create(99, 0, 1).getFullYear()).toBe(99);
+    expect(adapter.create(100, 0, 1).getFullYear()).toBe(100);
+  });
+
+  it('should create Date with low year number and over/under-flow', () => {
+    expect(adapter.create(50, 12 * 51, 1).getFullYear()).toBe(101);
+    expect(adapter.create(50, 12, 1).getFullYear()).toBe(51);
+    expect(adapter.create(50, -12, 1).getFullYear()).toBe(49);
+    expect(adapter.create(50, -12 * 51, 1).getFullYear()).toBe(-1);
+  });
+
+  it('should get today\'s date', () => {
```

Comment:
optional: you can use double quotes (or template string backticks) when it's to avoid escaping single quotes